nilearn/image/image.py

```diff
@@ -502,6 +502,61 @@ def swap_img_hemispheres(img):
 
     # create swapped nifti object
     out_img = nibabel.Nifti1Image(img.get_data()[::-1], img.get_affine(),
-        header=img.get_header())
+                                  header=img.get_header())
 
     return out_img
+
+
+def _index_img(imgs, index):
+    """Helper function for index_img and iter_img."""
+    return nibabel.Nifti1Image(imgs.get_data()[:, :, :, index],
+                               imgs.get_affine(),
+                               header=imgs.get_header())
+
+
+def index_img(imgs, index):
+    """Indexes into a 4D Niimg-like object in the fourth dimension.
+
+    Common use cases include extracting a 3D image out of `imgs` or
+    creating a 4D image whose data is a subset of `imgs` data.
+
+    Parameters
+    ----------
+    imgs: 4D Niimg-like object
+          See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+
+    index: Any type compatible with numpy array indexing
+           Used for indexing the 4D data array in the fourth dimension.
+
+    Returns
+    -------
+    output: nibabel.Nifti1Image
+
+    """
+    imgs = check_niimgs(imgs)
```

Comment:
> There could be 2 modes: browsing (in that case a list of 3D files will not be
> loaded, or just memmapped) and computing (in that case the whole thing is
> loaded).

Modal functions are not good for API. Users tend to get them wrong.

Aside from making the code simpler, I don't see a benefit of the greedy
loading.

> However, is this a real usecase? I have the impression that this is
> more for plotting maps rather than indexing scans in a whole sequence.

It's for everywhere. One example: if I need to load a resting-state data,
but discard the first 3 images.